### DISCUSSION TITLE
using mscred package

### DISCUSSION THREAD

**Discussion by lin00025** [iraf-community/iraf#188] [Q&A] [answered]
Hello,  

I have installed iraf-2.16.1 on MacOS Catalina v10.15, following the guide : https://iraf-community.github.io/install.html 
after that I need to use the external package mscred: https://github.com/iraf-community/iraf-mscred
I simply untar the file and place the directory in "/iraf/extern/", and now I can see mscred when I execute iraf.  

While when I try to apply any task in the package (say ccdproc for example),  it gives me the message: 
**ERROR on line 26: Cannot open connected subprocess (./x_mscred.e)**

I don't have the x_mscred.e in the iraf/bin or iraf/bin.macintel.  

Thank you,

**Comment by olebole** [comment, marked as answer]
Unfortunately, just untarring is not enough; you still need to compile it. And the external packages are currently nor really supported (it is unclear which of them is still in real use). What could work is:
```shell
cd $iraf/extern/mscred
mkdir bin.macintel
ln -s -f bin.macintel bin
export mscred=`pwd`/
mkpkg -p mscred
```
This should create a file `x_mscred.e` in `$iraf/extern/mscred/bin.macintel`.
Note that the package should be untarred to `$iraf/extern/mscred`, so if IRAF is in `/iraf/iraf`, this would be `/iraf/iraf/extern/mscred`.
If this way works, please give a feedback.

**Reply by lin00025** [reply]
Thanks for the reply!

I tried the above but didn't successfully get x_mscred.e in the bin.macintel.  
The first few lines shows  (/Users/linyuheng/ is my home directory): 
mscred $mkpkg -p mscred

```
envinit: cannot open `/Users/linyuheng/iraf-2.16.1-2018.11.01/extern/mscredlib/zzsetenv.def'
Warning, mkpkg line 0: cannot open `mscred$lib/mkpkg.inc'
check out file `libmscred.a = mscbin/libmscred.a'
warning: library `libmscred.a' not found
file list contains only object files
ar r /Users/linyuheng/iraf-2.16.1-2018.11.01/extern/mscred/src/mscbin/libmscred.a curfit.o t_curfit.o
ar: /Users/linyuheng/iraf-2.16.1-2018.11.01/extern/mscred/src/mscbin/libmscred.a: No such file or directory
file list contains only object files
ar r /Users/linyuheng/iraf-2.16.1-2018.11.01/extern/mscred/src/mscbin/libmscred.a imsurfit.o pixlist.o ranges.o t_imsurfit.o
ar: /Users/linyuheng/iraf-2.16.1-2018.11.01/extern/mscred/src/mscbin/libmscred.a: No such file or directory
```
I don't have the mscbin/ in mscred/src/, but the libmscred.a file is in mscred/src/

and the last few lines: 

```
check out file `x_mscred.o = mscbin/x_mscred.o'
xc -c -w x_mscred.x
xc -Nz  x_mscred.o -lmscred -lxtools -lcurfit   -lgsurfit -lslalib -lsurfit -liminterp -o xx_mscred.e
ld: library not found for -lmscred
clang: error: linker command failed with exit code 1 (use -v to see invocation)
check file `x_mscred.o' into `mscbin$'
move `xx_mscred.e' to `mscbin$x_mscred.e'
$move: file `xx_mscred.e' not found
Warning, mkpkg line 24: error moving file xx_mscred.e
Warning, mkpkg line 12: module `install' not found or returned error
Warning, mkpkg line 11: module `update@src' not found or returned error
```

**Reply by olebole** [reply]
Oh, the problem was that you need to specify a slash `/` at the end of the `mscred` environment variable:
```shell
cd $iraf/extern/mscred
mkdir bin.macintel
ln -s -f bin.macintel bin
export mscred=`pwd`/      #  <====
mkpkg -p mscred
```

**Reply by lin00025** [reply]
Hello,  
I think I mess up with the source code in the beginning.
After git clone the code again and add the "/" at the end, it is working now. 

Thank you!
